jetty-websocket/javax-websocket-client/src/test/java/examples/SecureWebSocketContainerExample.java

```diff
@@ -92,7 +92,7 @@ public static void main(String[] args)
     public static WebSocketContainer getConfiguredWebSocketContainer() throws Exception
     {
         URL jettyHttpClientConfigUrl = Thread.currentThread().getContextClassLoader()
-            .getResource("examples/jetty-websocket-httpclient.xml");
+            .getResource("jetty-websocket-httpclient.xml");
```

Comment:
The `XmlBasedHttpClientProvider` could not find the xml file when it was in `examples/` but worked when moved up a directory to resources, so I think that the ClassLoader scope stuff below doesn't work.

If you replace this entire method with just `ContainerProvider.getWebSocketContainer()` it works as well.

Reply:
Turns out the reason the xml does not work in `src/test/resources/examples` is because in jetty-10 we lazily create the HttpClient so that the `HttpClientProvider` is run during the call to `client.connectToServer(echoEndpoint, clientEndpointConfig, echoUri);` in main and not in the `ThreadClassLoaderScope` try block below.